client/src/docs-ui/page/page.tsx

```diff
@@ -34,6 +34,8 @@ import {parseURL} from "../../utils/url/url.worker";
 
 const SELECTED_TABS_LOCAL_STORAGE_KEY = `amplify-docs::selected-tabs`;
 
+/** increments every time the page changes, to tell all block switchers on the page to rerender */
+let alwaysRerenderBlockSwitcher = 0;
```

Comment:
Do we want to make this a `private` variable?

Reply:
Not quite sure what you meant by this, but I just pulled it into the class instead of leaving it global; hopefully that resolves it!